BEP100.md

```diff
@@ -0,0 +1,154 @@
+
+## 3. Status
+
+This BEP is a draft.
+
+## 4. Motivation
+
+Today BSC is experiencing network scalability problems and Binance have proposed to use BAS in their Outlook 2022 paper to solve this problem because these side chains can be designed for much higher throughput and lower gas fees. In this document we want to define a protocol for consensus management and messaging between BAS and BSC so that it is easier for developers to use a ready-made solution and it is easier for BSC to integrate with them.
+
+## 5. Specification
+
+Literally BAS is a modular framework for creating BSC-compatible side chains that defines requirements for integration with the BSC ecosystem and brings development-ready EVM-compatible features like staking, RPC-API or smart contracts. Since BSC doesn’t rely on the BAS security model, that is why there is no default embedded production-ready bridge solution between BSC and BAS networks, but instead of it BAS can provide protocols and standards for integrating third party bridges that can be managed by the BAS validator set of other projects like AnySwap or Celer Network cBridge, of course if they trust the BAS developer team.
+
+BAS brings several programmable and configurable modules that can be used or modified by developers to reach their business goals.
+
+Here is an example of such modules:
++ Networking - for p2p communication between different BAS nodes;
++ Blockchain & EVM - for block producing and EVM transaction execution, of course each BAS can define their own runtime execution environment based, for example, on WebAssembly;
++ Web3 API - for basic compatibility with Web3 ecosystem including MetaMask and other applications;
++ Transaction Pool - for managing internal BAS policies for transaction filtering and for charging fees for the system operational;
++ PoA & PoS Consensus - for users to be able to vote for the honest validators in the BAS network and guarantee the safeness of actions applied on the chain;
++ Storage & State - for persisting local data.
+
+### 5.0 Circulation Model and Native Asset Bridge
+
+The key parts of each BAS application are native token circulation model and cross chain bridge for native assets. Native assets of the BAS chain are located in the BAS application and managed by sidechain directly. BAS is designed to provide cross chain functionality for the native assets. Since native assets are fully managed by BAS developers they can compromise token supply or mint/burn tokens. We leave such a decision on the conscience of the developers, because by manipulating the supply of their tokens they only risk their reputation. The only thing required for native cross chain bridge is block header verification that allows one to keep an active validator set and verify the correctness of cross chain transactions.
```

Comment:
If BAS validators will be the relayers, the `block header verification` is not a must. Since we have L2 Zk rollup as a more promising option, we can leave this part aside first.

Reply:
Yes, you're right. But ZK rollup verification can also be implemented as block header verification function. Maybe it's better to rename it to just `verification function` to remove association with block headers.

Reply:
Of course its overhead to upload n/2+1 blocks, but instead of it we can use BN256 aggregated signatures with merkle patricia trie proof verification to proof the existence of data in the BAS sidechain.